.github/workflows/release.yaml

```diff
@@ -0,0 +1,56 @@
+name: Production Release (Docker Tag & Push)
```

Comment:
This file is more or less identical to cd.yaml, if I see this correctly. Is there an easy way of deduplicating the code? If it's not easy, I'm not asking you to spend a ton of time on figuring out how to do this. :)

Reply:
sure, the issue is in the trigger separation. happy to talk through what i've tried wrt this in the past with you at some point, bc i could be missing some other solution here. tl;dr, that "Build and Push to ECR" step is tricky to have do one thing only for releases and another thing only for commits to main branch (push or merge), because of some incompleteness of docs around triggers at the step(s) level instead of the whole pipeline level.

i hope that made sense.

Reply:
so this is the conditional (what you can use at the step level) docs: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions

and this is the context where i had issues pinning down filtering the github.ref for a branch name versus what string to match on for it being any release: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

Reply:
After looking at that documentation for a bit, I believe that the trigger at the top of this file is equivalent to the condition

    if: github.event_name == 'release' && contains(['published', 'created', 'edited'], github.event.action)

See the [documentation of the event payload for a `release` event](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release). I didn't test this at all, though, so chances are it's wrong. :)

Again, I'm not requesting any changes here – feel free to either try the condition above, or merge it as it is, and we'll deal with the ocde duplication in case it ever becomes annoying for some reason.

Reply:
sure, i can revisit. i did try something similar, but lets see.

Reply:
also, those docs have updated in the last 2-3 weeks it appears: 
![image](https://user-images.githubusercontent.com/6874040/127651314-e3b6b343-bd2f-4378-a2e8-09e4636bf276.png)